### ISSUE TITLE
[BUG] frame-ancestors directive in CSP causing refuse to load browser console error

### ISSUE THREAD

**Issue by AdmSteele** [dm-p/powerbi-visuals-html-content#36]
**Describe the bug**

With a Content Security Policy (CSP) enabled and a frame-ancestors directive such as `frame-ancestors 'self' 'unsafe-inline' https://* ms-appx-web://*;` (which should allow everything via the https://* item), the browser still refuses to load the iframe and the browser console returns the error: 
`Refused to load https://run.outfieldapp.com because it does not appear in the frame-ancestors directive of the Content Security Policy.`

When the CSP is disabled it loads successfully.

When the CSP is enabled and you try to load it with an iframe html snippet in a jsfiddle it loads successfully.

**To Reproduce**

1. Our embed via power bi is behind a user+password wall so I cannot share a direct example.
2. However, this CSP is active on https://run.outfieldapp.com/users/sign_in for example so you should be able to make on quickly in power bi yourself and reproduce it using the URL in an html content iframe and then try it in a browser (not IE).

**Expected behavior**

Expected iframe to load.

**Screenshots/Workbook**

n/a

**Power BI Setup (please complete the following information):**
 - Platform: Power BI Desktop
 - OS: windows 10
 - Browser Safari or Chrome
 - Version Safari Version 14.0.3 (16610.4.3.1.4) or Chrome Version 89.0.4389.114 (Official Build) (x86_64)

**Visual Information (please complete the following information, available by right-clicking the visual in Power BI and selecting *About*):**
 - Version: it's a customer's installation. I will see if I can get this.

**Additional context**

n/a

**Comment by dm-p** [comment]
Hi @AdmSteele  - I don't know much about CSPs, but from a very short orientation, I believe that because the developer visual host (which Power BI uses for all custom visuals) is itself a sandboxed iframe with `allow-same-origin` disabled, and I believe this will cause a violation of the CSP.

If you've not read it, please take a look at [this section of the Limitations page on the visual's website](https://www.html-content.com/reference/limitations#custom-visuals-high-level) and advise if this sounds like the kind of situation you're in. If it is, then unfortunately this is a limitation that is imposed by MS and there's not much we can do about it, unfortunately.

**Comment by AdmSteele** [comment]
@dm-p Yes I had read this on your repo. It may be the issue going on here. It seems to me that a CSP is independent of that, but I do not know for sure. Thanks for taking a look. Please let me know if you learn more.

**Comment by AdmSteele** [comment]
Update: when I removed the frame-ancestors line all together from the CSP, the iframe loads successfully. But the CSS is still blocked due to the style-src line in the CSP, since the now removed frame-ancestors line is not blocking it all together. I think it's safe to say the root issue is an incompatibility between HTML Content / PowerBI and CSPs since neither the frame-ancestors line nor the style-src line are being handled normally and when you remove those lines one at a time then it starts working. 

At least I can use this workaround for now to keep our CSP live and get a non-styled page to load in the iframe so the customer can at least see the content.
